### PR TITLE
Haddock web

### PR DESCRIPTION
**Pull request by lsmor**
Update the buildDocumentation.md to add a section for http launching. This allows the browser to run JavaScript, which I've found important in order to use haddock's search utility (very helpfull to find out modules and functions). The instructions can be helpfull  for the Community Playground which currently doesn't have built-in docs

### CODE REVIEW

**Comment by lsmor** [comment]
For some reason all commits since file creation are shown, but only two last commits are new

**Comment by dino-** [comment]
@lsmor This is awesome! Shortly I will try your instructions on my own dev system to be thorough.
I would love to see your Servant app here added to my `plutus-pioneer-systemd` project so that the Haddock path is parameterized somehow and it gets started/stopped/restarted with a new `plutus-api-docs` service. I'll think about how this might work.

File: book/docs/setup/buildDocumentation.md [hunk 1]
```diff
@@ -25,13 +25,84 @@ brave-browser ~/plutus/result/share/doc/index.html
...
+main :: IO ()
+main = do
+    putStrLn "running plutus documentation in http://localhost:8080/plutus-haddock/index.html"
+    run 8080 app
+```
+
+- Compile the file directly with `ghc` and run the executable (once compiled, it can run from outside the `nix-shell`)
+
+```bash
+[nix-shell:path/to/haddock-web] > ghc main.hs
```
Comment (dino-) [inline book/docs/setup/buildDocumentation.md]: Minor but I do like explicitly named binaries: `> ghc -o plutus-haddock main.hs`

If this seems too picky, please ignore!

File: book/docs/setup/buildDocumentation.md [hunk 2]
```diff
@@ -25,13 +25,84 @@ brave-browser ~/plutus/result/share/doc/index.html
...
+[1 of 1] Compiling Main             ( main.hs, main.o )
+Linking main ...
```
Comment (dino-) [inline book/docs/setup/buildDocumentation.md]: To match above: `+ Linking plutus-haddock ...`

File: book/docs/setup/buildDocumentation.md [hunk 3]
```diff
@@ -25,13 +25,84 @@ brave-browser ~/plutus/result/share/doc/index.html
...
+[nix-shell:path/to/haddock-web] > ./main # nix-shell not necessary
```
Comment (dino-) [inline book/docs/setup/buildDocumentation.md]: Again: `> ./plutus-haddock # nix-shell not necessary`

File: book/docs/setup/buildDocumentation.md [hunk 4]
```diff
@@ -25,13 +25,84 @@ brave-browser ~/plutus/result/share/doc/index.html
...
+      Application,
+      Raw,
+      type (:>),
+      serve,
+      serveDirectoryWebApp )
+import Data.Kind ()
+import Network.Wai.Handler.Warp (run)
+
+type Haddock = "plutus-haddock" :> Raw
+
```
Comment (dino-) [inline book/docs/setup/buildDocumentation.md:69]: Port 8080 is clashing with my Plutus Playground server, we should use a different one. Suggesting you add a variable for the port so the output matches, for the next comment suggestion:

```
serverPort :: Int
serverPort = 8081

```

File: book/docs/setup/buildDocumentation.md [hunk 5]
```diff
@@ -25,13 +25,84 @@ brave-browser ~/plutus/result/share/doc/index.html
...
+myApi :: Proxy Haddock
+myApi = Proxy
+
+app :: Application
+app = serve myApi server
+
+main :: IO ()
+main = do
+    putStrLn "running plutus documentation in http://localhost:8080/plutus-haddock/index.html"
+    run 8080 app
```
Comment (dino-) [inline book/docs/setup/buildDocumentation.md]: And then to go with the `serverPort` variable above, suggesting this:

```
    putStrLn $ "running plutus documentation in http://localhost:"
      <> (show serverPort) <> "/plutus-haddock/index.html"
    run serverPort app
    ```

File: book/docs/setup/buildDocumentation.md (hunk 1 above)
Comment (lsmor) [reply book/docs/setup/buildDocumentation.md]: I do agree, It is better an explicit binary name

File: book/docs/setup/buildDocumentation.md (hunk 4 above)
Comment (lsmor) [reply book/docs/setup/buildDocumentation.md:69]: :+1:

**Comment by lsmor** [comment]
@dino- your improvements have been commited :) . 

With respect to add this to `plutus-pioneer-systemd` I think is a great Idea too, but I am not so familiar with `systemctl` so probably I can't help a lot. I guess it should be easy once you have the executable. Any case, don't hesitate to ask me or propose any improvements so It can work better with your launching script. 

One major issue is that the haddock path is hardcoded and it changes every time the plutus-core commit is pumped. So, if you want to avoid recompilation, It'd need to be parse and passed as a parameter. I might work on that tomorrow

**Comment by dino-** [comment]
I think this is ready to merge.

**Comment by nstankov-bg** [comment]
Thank you!

**Pull request merged**